kooplearn/nn/functional.py

```diff
@@ -146,70 +148,42 @@ def deepprojection_score(
         S = torch.trace(M_X @ M_Y)
     return S - 0.5 * metric_deformation * (R_X + R_Y)
 
-
-def eym_score_Ustat(
-    X: torch.Tensor,
-    Y: torch.Tensor,
-    metric_deformation: float = 1.0,
-    center: bool = True,
-):
-    if center:
-        _X = X - X.mean(0, keepdim=True)
-        _Y = Y - Y.mean(0, keepdim=True)
-    else:
-        _X = X
-        _Y = Y
-
-    joint_measure_score = ((_X * _X).sum(dim=-1)).mean() + (
-        (_Y * _Y).sum(dim=-1)
-    ).mean()
-    product_measure_score = torch.square_(_X @ _Y.T)
-    product_measure_score = product_measure_score.fill_diagonal_(
-        0
-    )  # Zeroing out the diagonal in place
-    b = X.shape[0]
-    product_measure_score = b * product_measure_score.mean() / (b - 1)
-    score = joint_measure_score - product_measure_score
-    return score
-
-
-def eym_score_split(
+    
+def eym_score(
     X: torch.Tensor,
     Y: torch.Tensor,
-    metric_deformation: float = 1.0,
+    metric_deformation: float = 0.0,
+    center_covariances: bool = True,
 ):
-    if X.shape[0] // 2 != 0:
-        X = X[:-1]
-        Y = Y[:-1]
-    batch_size = X.shape[0]
-    # Randomly splitting in half
-    idxs = torch.randperm(batch_size)
-    U1 = X[idxs[: batch_size // 2]]
-    U2 = X[idxs[batch_size // 2 :]]
-
-    V1 = Y[idxs[: batch_size // 2]]
-    V2 = Y[idxs[batch_size // 2 :]]
-
-    # centered covariance matrices
-    cov_U1 = torch.cov(U1.T)
-    cov_U2 = torch.cov(U2.T)
-    cov_V1 = torch.cov(V1.T)
-    cov_V2 = torch.cov(V2.T)
-
-    cov_U1V1 = cross_covariance(U1.T, V1.T, centered=True)
-    cov_U2V2 = cross_covariance(U2.T, V2.T, centered=True)
-
-    score = -1 * (
-        0.5 * (torch.trace(cov_U1 @ cov_V2) + torch.trace(cov_U2 @ cov_V1))
-        - torch.trace(cov_U1V1)
-        - torch.trace(cov_U2V2)
-    )
-    if score > 0:
-        d = U1.shape[-1]
+    """Eckart-Young-Mirsky (EYM) score by [unpublished].
+
+    Args:
+        X (torch.Tensor): Covariates for the initial time steps.
+        Y (torch.Tensor): Covariates for the evolved time steps.
+        metric_deformation (float, optional): Strength of the metric metric deformation loss: Defaults to 0.0.
+        center_covariances (bool, optional): Use centered covariances to compute the EYM score. Defaults to True.
+
+    Returns:
+        torch.Tensor: EYM score
+    """
+    U1, U2, V1, V2 = random_split(X, Y, 2)
+
+    cov_U1 = covariance(U1, center=center_covariances)
+    cov_U2 = covariance(U2, center=center_covariances)
+    cov_V1 = covariance(V1, center=center_covariances)
+    cov_V2 = covariance(V2, center=center_covariances)
+
+    cov_U1V1 = cross_covariance(U1, V1, rowvar=False, center=center_covariances)
+    cov_U2V2 = cross_covariance(U2, V2, rowvar=False, center=center_covariances)
+
+    score = 0.5 * (torch.sum(cov_U1*cov_V2) + torch.sum(cov_U2*cov_V1)) - torch.trace(cov_U1V1) - torch.trace(cov_U2V2)
```

Comment:
Just nit-picking, but the name of this variable should be "loss" (something to be minimized) and not "score" (something to be mazimized). Am I right? 

The name of the function is eym_score, so it returns something that it should be maximized.

Reply:
You are right and I saw your 'Updated docs' commit. Do you prefer to keep the consistency with `vamp_score` and `deepprojection_score`, as you've done, or should we consider renaming `eym_score` function (and variable therein) to `loss` for clarity? 'Loss' might be more intuitive, especially since the NCP paper also refers to it in those terms.

Reply:
Just rename the variable within to `loss` and we'll be consistent with the rest of the package